spring-integration-core/src/main/java/org/springframework/integration/handler/LoggingHandler.java

```diff
@@ -164,58 +169,66 @@ protected void onInit() throws Exception {
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
 		switch (this.level) {
-		case FATAL:
-			if (this.messageLogger.isFatalEnabled()) {
-				this.messageLogger.fatal(createLogMessage(message));
-			}
-			break;
-		case ERROR:
-			if (this.messageLogger.isErrorEnabled()) {
-				this.messageLogger.error(createLogMessage(message));
-			}
-			break;
-		case WARN:
-			if (this.messageLogger.isWarnEnabled()) {
-				this.messageLogger.warn(createLogMessage(message));
-			}
-			break;
-		case INFO:
-			if (this.messageLogger.isInfoEnabled()) {
-				this.messageLogger.info(createLogMessage(message));
-			}
-			break;
-		case DEBUG:
-			if (this.messageLogger.isDebugEnabled()) {
-				this.messageLogger.debug(createLogMessage(message));
-			}
-			break;
-		case TRACE:
-			if (this.messageLogger.isTraceEnabled()) {
-				this.messageLogger.trace(createLogMessage(message));
-			}
-			break;
-		default:
-			throw new IllegalStateException("Level '" + this.level + "' is not supported");
+			case FATAL:
+				if (this.messageLogger.isFatalEnabled()) {
+					this.messageLogger.fatal(createLogMessage(message));
+				}
+				break;
+			case ERROR:
+				if (this.messageLogger.isErrorEnabled()) {
+					this.messageLogger.error(createLogMessage(message));
+				}
+				break;
+			case WARN:
+				if (this.messageLogger.isWarnEnabled()) {
+					this.messageLogger.warn(createLogMessage(message));
+				}
+				break;
+			case INFO:
+				if (this.messageLogger.isInfoEnabled()) {
+					this.messageLogger.info(createLogMessage(message));
+				}
+				break;
+			case DEBUG:
+				if (this.messageLogger.isDebugEnabled()) {
+					this.messageLogger.debug(createLogMessage(message));
+				}
+				break;
+			case TRACE:
+				if (this.messageLogger.isTraceEnabled()) {
+					this.messageLogger.trace(createLogMessage(message));
+				}
+				break;
+			default:
+				throw new IllegalStateException("Level '" + this.level + "' is not supported");
 		}
 	}
 
-
 	private Object createLogMessage(Message<?> message) {
+		Assert.notNull(this.expression, "'expression' must not be null");
```

Comment:
That's exactly I've written you about :smile: Let's implement this solution and get rid of `onInit` method at all.